### PR TITLE
DAOS-13940 gurt: Fix log rotation error

### PR DESCRIPTION
**Pull request by frostedcmos**
- When rotating a log that has stderr merged into it, the log was reopened but a new fd was not retreived, resulting in old fd sometimes being bad.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by soumagne — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

File: src/gurt/dlog.c [hunk 1]
```diff
@@ -387,6 +387,77 @@ log_exceed_threshold(void)
...
 }
 
+static int
+log_rotate(void)
+{
+	int rc = 0;
+
+	/* exceeds the size threshold, rename the current log file
+	 * as backup, create a new log file.
+	 */
```
Comment (ashleypittman) [inline src/gurt/dlog.c:397]: This is a comment for the new function so goes on L389.

File: src/gurt/dlog.c [hunk 2]
```diff
@@ -387,6 +387,77 @@ log_exceed_threshold(void)
...
+		close(mst.log_old_fd);
+		mst.log_old_fd = -1;
+	}
+
+	/* remove the backup log file */
+	rc = unlink(mst.log_old);
+	if (rc && errno != ENOENT) {
+		dlog_print_err(errno, "failed to unlink old file\n");
+		return -1;
+	}
```
Comment (ashleypittman) [inline src/gurt/dlog.c:416]: You don't need to do this, the rename will clobber any existing file.

Comment (frostedcmos) [reply src/gurt/dlog.c]: Removed

File: src/gurt/dlog.c (hunk 1 above)
Comment (frostedcmos) [reply src/gurt/dlog.c]: Moved the comment to the function start

**Review by soumagne — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Pull request merged**